core/electron/src/frontend/ElectronApp.ts

```diff
@@ -5,15 +5,22 @@
 import { ProcessDetector, PromiseReturnType } from "@itwin/core-bentley";
 import { IpcListener, IpcSocketFrontend } from "@itwin/core-common";
 import { IpcApp, NativeApp, NativeAppOpts } from "@itwin/core-frontend";
-import { ITwinElectronApi } from "../backend/ElectronPreload";
+import type { IpcRenderer } from "electron";
 import { DialogModuleMethod } from "../common/ElectronIpcInterface";
 import { ElectronRpcManager } from "../common/ElectronRpcManager";
+import type { ITwinElectronApi } from "../common/ITwinElectronApi";
+
+declare global {
+  interface Window {
+    itwinjs: ITwinElectronApi;
+  }
+}
 
 /**
  * Frontend Ipc support for Electron apps.
  */
 class ElectronIpc implements IpcSocketFrontend {
-  private _api: ITwinElectronApi;
+  private _api: ITwinElectronApi | IpcRenderer;
```

Comment:
Why did you need to IpcRenderer here?

Reply:
thats the type when window.itwinjs doesn't exist,

```ts
this._api = window.itwinjs ?? require("electron").ipcRenderer;
```

Reply:
ah okay, missed that before.